astropy/units/core.py

```diff
@@ -1968,6 +2027,16 @@ class CompositeUnit(UnitBase):
         A sequence of powers (in parallel with ``bases``) for each
         of the base units.
     """
+    def __new__(cls, scale=None, bases=None, powers=None, decompose=False,
+                decompose_bases=set(), _error_check=True):
+        # Have to set defaults of None since we can enter here from picking.
```

Comment:
Thanks for pointing that out. Obviously, I didn't know about pickling enough, and should have read the documentation!